CHANGES.txt

```diff
@@ -31,6 +31,21 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
       determination when configuring the build environment.  This could lead to build failures when
       only an MSVC Express instance is installed and the MSVC version is not explicitly specified
       (issue #2668 and issue #2697).
+    - The imported system environment variable names for MSVC 7.0 and 6.0 have been changed to the
+      names set by their respective installers.  Prior to this change, bypassing MSVC detection by
+      specifying the MSVC 7.0 batch file directly would fail due to using an erroneous environment
+      variable name.  Arguments are no longer passed to the MSVC 6.0 to 7.1 batch files as no
```

Comment:
I don't see any code which has this change?

Reply:
MSCommon/common.py:
```
'VS70COMNTOOLS' (does not exist) changed to 'VSCOMNTOOLS' (set by installer)
'VS60COMNTOOLS' (does not exist) changed to 'MSDevDir' (set by installer but is not common tools dir)
```

Reply:
This big `Arguments are no longer passed to the MSVC 6.0 to 7.1 batch files as no` ?

Reply:
Sorry.

It is an artifact of the refactoring: the msvc batch file argument string is empty for 6.0-7.1.

```
...
    arg = ''
    if vernum > 14:
        # 14.1 (VS2017) and later
        batfiledir = os.path.join(pdir, "Auxiliary", "Build")
        batfile, _ = _GE2017_HOST_TARGET_BATCHFILE_CLPATHCOMPS[(host_arch, target_arch)]
        batfilename = os.path.join(batfiledir, batfile)
    elif 14 >= vernum >= 8:
        # 14.0 (VS2015) to 8.0 (VS2005)
        arg, _ = _LE2015_HOST_TARGET_BATCHARG_CLPATHCOMPS[(host_arch, target_arch)]
        batfilename = os.path.join(pdir, "vcvarsall.bat")
    else:
        # 7.1 (VS2003) and earlier
        pdir = os.path.join(pdir, "Bin")
        batfilename = os.path.join(pdir, "vcvars32.bat")
...
```

Reply:
? but it wasn't set to any values for the previous logic
```
    elif 8 > vernum >= 7:
        # 7.1 (VS2003) to 7.0 (VS2003)
        pdir = os.path.join(pdir, os.pardir, "Common7", "Tools")
        batfilename = os.path.join(pdir, "vsvars32.bat")
``` 

Only set for vernum between 14 and 8?

Reply:
It looks like the message probably should have been included in #4131.  Prior to 4131, there was a boolean use_arg value that was returned.  The refactoring returned the argument string rather than the boolean value.  There were conflicts that had to be resolved between this PR and 4131.  It looks like the text was written expecting this PR to be accepted before 4131.

Reply:
> Only set for vernum between 14 and 8?

Yes.

The newer msvc batch files (14.1 and later) use the host/target specific batch files (e.g., ```vcvarsamd64_x86.bat```).
The middle msvc batch files (14 to 8) need the host/target argument (e.g., ```amd64_x86``` to vcvarsall.bat).
The older batch files (7.1 and earlier) don't take any arguments.

Reply:
@bdbaddog Confirmed that the suppressing the arguments is a byproduct of the refactoring in #4131.  

Prior to 4132, the argument was always set to the host/target string and the a boolean was used that indicated where the argument should be used or not (reset to empty).  The refactoring eliminated the boolean variable and set the argument as needed.

The first commit in this thread suppressed the argument by setting the boolean use arg value to False.

The changes.txt blurb for suppressing the argument should have been included in #4131.   I knew there were conflicts with the two branches and how to resolve appropriately.  It appears that I was relying on this PR to be accepted before 4131.  

Sorry.  Good catch on the review.

How would you like to proceed?

Reply:
ok. that's fine. if the blurb is really just catching up something which was already merged.

So it's good to merge then?

Reply:
Yes.  Good to merge.